userspace/libsinsp/test/events_file.ut.cpp

```diff
@@ -679,3 +679,33 @@ TEST_F(sinsp_with_test_input, test_fdtypes)
 	ASSERT_EQ(get_field_as_string(evt, "fd.types[3]"), "(bpf)");
 	ASSERT_EQ(get_field_as_string(evt, "fd.types"), "(bpf,file)");
 }
+
+TEST_F(sinsp_with_test_input, test_pidfd)
+{
+	add_default_init_thread();
+	open_inspector();
+	sinsp_evt* evt = NULL;
+	int64_t pidfd = 1;
+	int64_t pid = 2;
+	int64_t target_fd = 3;
+	int64_t fd = 4;
+
+	/* Open a file descriptor */
+	add_event_advance_ts(increasing_ts(), pid, PPME_SYSCALL_OPEN_X, 6, (uint64_t)target_fd, "/tmp/the_file", PPM_O_RDWR, 0, 5, (uint64_t)123);
+
+
+	/* Create a pidfd using the same pid */
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_PIDFD_OPEN_X, 3, pidfd, pid, 0);
+	
+	ASSERT_EQ(evt->get_type(), PPME_SYSCALL_PIDFD_OPEN_X);
+	ASSERT_EQ(get_field_as_string(evt,"fd.num"), std::to_string(pidfd));
```

Comment:
Since we're adding a new type of FD, let's also test the correct value for `fd.type` and `fd.typechar` both here and in the event below.